packages/container/libraries/strongly-typed/src/container.ts

```diff
@@ -1,22 +1,33 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { Container, type interfaces } from 'inversify';
+import type { Newable, ServiceIdentifier } from '@inversifyjs/common';
+import type {
+  BindToFluentSyntax,
+  ContainerOptions,
+  IsBoundOptions,
+} from '@inversifyjs/container';
+import { Container } from '@inversifyjs/container';
+import type {
+  BindingActivation,
+  BindingDeactivation,
+  GetOptions,
+} from '@inversifyjs/core';
```

Comment:
I would hazard to say these types are exposed in `inversify@7`. I prefer to rely on `inversify@7` since `@inversify/*` dependencies are internal ones and major releases are released way more often.

Reply:
I originally tried to do this but I think I'm missing the `BindingActivation` and `BindingDeactivation` types.

Reply:
I've rebased onto `main`. Let me know what you want me to do about these types. Options:

- expose them in `inversify` (probably best option? For this library at least)
- hand-roll the type to be compatible (fragile, but doesn't require a change to `inversify`)
- add a dependence on `@inversifyjs/core` (but you said you don't want to do that)
- other...?

Reply:
I see, yeah, `BindingActivation` and `BindingDeactivation` must be exposed in `inversify`. I'll do it during the weekend.